### ISSUE TITLE
Linux: MultiplayerSample project fails to build 

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de-multiplayersample#303]
**Describe the bug**
MultiplayerSample project fails to build after following https://github.com/o3de/o3de-multiplayersample/blob/stabilization/2305/README_LINUX.md steps. The error occurs both when built via Project Manager and from the Terminal (using commands from here: https://github.com/o3de/o3de-multiplayersample/blob/stabilization/2305/README_LINUX.md#step-3-configure-and-build).

It should be noted that this issue occurred on 2/2 machines when testing with installer and on 1/2 machines when testing with GH O3DE build (Pre-built SDK).

It should also be noted that this issue occurred only on Linux.

**Steps to reproduce**
1. Follow steps listed here: https://github.com/o3de/o3de-multiplayersample/blob/stabilization/2305/README_LINUX.md.

**Expected behavior**
MultiplayerSample project is built.

**Actual behavior**
MultiplayerSample project fails during the build after configuration.

**Failed project build log**

[CMakeProjectBuildError.log](https://github.com/o3de/o3de-multiplayersample/files/11148045/CMakeProjectBuildError.log)

**Found in Branch**
o3de-multiplayersample: Stabilization/2305 (531c59015d14a169619657457c09240acec1c3d0)
o3de-multiplayersample-assets: Stabilization/2305 (6739f1027152e68f67be361d77df2296dbf4e800)
o3de: Stabilization/2305 (c4dd0ce2cb370446c18a951296c306073ddcf193)

**Desktop**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 22.04.1 LTS
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by lmbr-pip** [comment]
```
/home/gpc-610/O3DE/Projects/o3de-multiplayersample/Gem/Code/Source/Components/Multiplayer/GemSpawnerComponent.cpp:188:33: error: loop variable 'gemType' creates a copy from type 'const MultiplayerSample::GemSpawnable' [-Werror,-Wrange-loop-construct]
        for (const GemSpawnable gemType : GetParent().GetGemSpawnables())
```

Linux code bug

**Comment by lmbr-pip** [comment]
Also needs https://github.com/o3de/o3de-multiplayersample/pull/311

**Comment by lmbr-pip** [comment]
Resolved as of 653daac21eb1392b6ecdf1702d9ae6cf463e9da8, branch should build.

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on o3de-multiplayersample stabilization/2305 (4cee1b11a04ecbedacadb7dd977f11c8adce0d12).
